### PR TITLE
CORE-711: In BRCryptoTransferState use BRCryptoFeeBasis

### PR DESCRIPTION
**Pull request by EBGToo**
'Draft' waiting on CORE-696.

### CODE REVIEW

File: Swift/BRCrypto/BRCryptoTransfer.swift [hunk 1]
```diff
@@ -321,7 +321,9 @@ public enum TransferState {
             confirmation: TransferConfirmation (blockNumber: core.u.included.blockNumber,
                                                 transactionIndex: core.u.included.transactionIndex,
                                                 timestamp: core.u.included.timestamp,
-                                                fee: core.u.included.fee.map { Amount (core: $0, take: false) }))
+                                                fee: core.u.included.feeBasis
```
Comment (michael-brd) [inline Swift/BRCrypto/BRCryptoTransfer.swift]: Leaking `feeBasis` here?

Comment (EBGToo) [reply Swift/BRCrypto/BRCryptoTransfer.swift]: Is it?  The caller to `init` owns `core` and thus `feeBasis`.  The `core.u.included.feeBasis` doesn't take the reference.  Am I missing it?

Comment (michael-brd) [reply Swift/BRCrypto/BRCryptoTransfer.swift]: The C code has taken a reference to the fee basis before passing it up to the Swift/Java layer. You then get the fee out of it but you don't give the feebasis reference up (that I notice, maybe I am missing it?).

Comment (EBGToo) [reply Swift/BRCrypto/BRCryptoTransfer.swift]: Addressed w/ cryptoTransferStateRelease() in the TransferState initializer.  Does that appear sufficient?

File: crypto/BRCryptoTransfer.c [hunk 2]
```diff
@@ -811,7 +811,7 @@ cryptoTransferStateRelease (BRCryptoTransferState *state) {
     switch (state->type) {
         case CRYPTO_TRANSFER_STATE_INCLUDED: {
             if (NULL != state->u.included.feeBasis) {
-                cryptoFeeBasisTake (state->u.included.feeBasis);
+                cryptoFeeBasisGive (state->u.included.feeBasis);
```
Comment (michael-brd) [inline crypto/BRCryptoTransfer.c]: Good catch.

**Pull request merged**